ilqgame_planner:
  frames:
    fixed: world

  topic:
    traj: /ucb/ilq_traj
    state:
      names:
        - /ucb/ego
        - /ucb/other
      types:
        - EgoState
        - OtherState

  viz:
    traj: /ucb/viz/ilq_traj

  replanning_interval: 0.1

  # goals:
  #   p1:
  #     x: 3.0
  #     y: 3.0
  #   p2:
  #     x: 3.0
  #     y: -3.0

  speed:
    min: 1.0
    max: 4.0
    nominal: 3.0 # 3.5 for 135 degree and 45
    dubins: 4.0 # 5 for 90 degree, 2 for 45

  heading:
    other:
      nominal: 0.0 # 0.78, -0.78, -1.57

  weight:
    ego:
      u:
        accel: 150000.0
        decel: 200000.0
        omega: 100000.0
      x:
        goal: 100.0
        proximity: 20.0 # 10 to induce ecic on 180 degree, 1 on 135
        v:
          max: 100000.0
          nominal: 10000.0
        lane:
          center: 20.0
          boundary: 100.0

    other:
      u:
        omega: 500000000.0
      x:
#        goal: 0.1
        proximity: 0.0 # HACK! Needs tuning.
        adversarial: 0.01 # HACK! Needs tuning.
        heading: 1.0 # HACK! Needs tuning.

  window:
    goal: 8.0
    adversarial: 1.0

  avoidance_margin:
    ego: 40.0 # HACK! Needs tuning.
    other: 40.0

  lane:
    half_width: 3.0

  srv:
    polyline_position: /polyline_position
